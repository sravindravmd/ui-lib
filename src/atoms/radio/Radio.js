import React, { useMemo } from 'react'
import PropTypes from 'prop-types'
import { v4 as uuidv4 } from 'uuid'
import { StyledRadio } from './Radio.styles'

export const Radio = ({
  label,
  value,
  checked,
  onChange,
  name,
  tabIndex = 0,
  disabled,
  className,
  ...other
}) => {
  const uid = useMemo(() => uuidv4(), [])

  return (
    <StyledRadio {...other} className={className}>
      <input
        type='radio'
        name={name || uid}
        className='radio-input'
        checked={checked || false}
        value={value}
        onChange={event => onChange(event)}
        id={uid}
        tabIndex={tabIndex}
        disabled={disabled}
      />
      <label htmlFor={uid} className='radio-label'>
        <span className='circle'></span>
        <span className='label'>{label}</span>
      </label>
      {/* <RadioFocus /> */}
    </StyledRadio>
  )
}

Radio.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  id: PropTypes.string,
  className: PropTypes.string,
  checked: PropTypes.bool,
  name: PropTypes.string,
  onChange: PropTypes.func,
  tabIndex: PropTypes.number,
  disabled: PropTypes.bool,
}

Radio.defaultProps = {
  checked: false,
}
