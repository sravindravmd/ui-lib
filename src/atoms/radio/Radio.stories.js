import React from 'react'
import { Radio } from './Radio'

export default {
  title: 'atoms/Radio',
  component: Radio,
  argTypes: {
    label: 'Label',
    value: null,
    checked: false,
    name: 'name',
    onClick: { actions: 'clicked' },
    tabIndex: 0,
  },
}

const Template = args => (
  <>
    <Radio {...args} />
    <Radio {...args} />
    <Radio {...args} />
  </>
)

export const Default = Template.bind({})
Default.args = {
  label: 'Label',
  value: 'default',
  checked: true,
  name: 'default',
  tabIndex: 0,
}
