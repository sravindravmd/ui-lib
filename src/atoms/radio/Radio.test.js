import React from 'react'
import { Radio } from './radio'
import {
  render,
  cleanup,
  fireEvent,
  waitFor,
  screen,
} from '@testing-library/react'
import 'regenerator-runtime/runtime'
import 'uuid'

afterEach(cleanup)
jest.mock('uuid', () => {
  return { v4: jest.fn(() => 1) }
})

const props = {
  label: 'Radio Label',
  value: 'default',
  checked: true,
  name: 'default',
  tabIndex: 0,
  id: 'radio',
}

describe('<Radio/>', () => {
  it('renders', () => {
    const { container } = render(<Radio {...props} />)
    expect(container).toMatchSnapshot()
  })

  it('check if radio is in dom', async () => {
    const { container } = render(<Radio {...props} />)
    await waitFor(() => screen.getByText(/Radio Label/i))
    const radio = container.querySelector('.radio-input')
    fireEvent.click(radio)
  })
})
