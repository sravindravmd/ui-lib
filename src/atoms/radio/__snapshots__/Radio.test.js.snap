// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio/> renders 1`] = `
<div>
  <div
    class="sc-bdfBwQ kVxuYK"
    id="radio"
  >
    <input
      checked=""
      class="radio-input"
      id="1"
      name="default"
      tabindex="0"
      type="radio"
      value="default"
    />
    <label
      class="radio-label"
      for="1"
    >
      <span
        class="circle"
      />
      <span
        class="label"
      >
        Radio Label
      </span>
    </label>
  </div>
</div>
`;
